NuRadioReco/detector/RNO_G/rnog_detector.py

```diff
@@ -1203,31 +1203,33 @@ def get_time_delay(self, station_id, channel_id, cable_only=False, use_stored=Tr
         signal_chain_dict = self.get_channel_signal_chain(
             station_id, channel_id)
 
-        time_delay = 0
-        for key, value in signal_chain_dict["response_chain"].items():
-
-            if re.search("cable", key) is None and re.search("fiber", key) and cable_only:
-                continue
-
-            weight = value.get("weight", 1)
-            if use_stored:
-                if "time_delay" not in value and "cable_delay" not in value:
-                    self.logger.warning(
-                        f"The signal chain component \"{key}\" of station.channel "
-                        f"{station_id}.{channel_id} has no cable/time delay stored... (hence return time delay without it)")
+        if use_stored and not cable_only:
+            resp = self.get_signal_chain_response(station_id, channel_id)
+            return resp.get_time_delay()
+        elif use_stored and cable_only:
+            time_delays = resp.get_time_delays()
+            names = resp.get_names()
+            time_delay = 0
+            for name, dt in zip(names, time_delays):
+                if re.search("cable", name) is None and re.search("fiber", name):
```

Comment:
shouldn't it be `if re.search("cable", name) is None and re.search("fiber", name) is None`?